invenio/modules/deposit/models.py

```diff
@@ -521,8 +553,29 @@ def __unicode__(self):
         """ Return a name for this class """
         return self.get_identifier()
 
+    @classmethod
+    def all_authorized(cls, current_user):
+        """
+        Return a dict of deposit types that the current user
+        is allowed to view.
+        """
+        from invenio.modules.access.control import acc_get_action_id, \
+            acc_find_possible_actions_user
+
+        all_types = cls.all()
+        action_id = acc_get_action_id('usedeposit')
+
+        avail_actions = acc_find_possible_actions_user(current_user.get_id(),
+                                                       action_id)[1:]
+        values = [s for s in itertools.chain(*avail_actions)
+                  if isinstance(s, basestring)]
+
+        # filter the all_types dict()
+        return {key: all_types[key] for key in values if key in all_types}
```

Comment:
Ah, I think I understand. So you want to have some kind of a wildcard - if a user is allowed to (usedeposit, type=None), this would be equivalent to a right to use all the types?

I can do that, but I think it would require additional `or s is None` in the if condition in the set comprehension.